firebase.go

```diff
@@ -173,6 +173,16 @@ func (fb *Firebase) doRequest(method string, body []byte) ([]byte, error) {
 	}
 
 	resp, err := fb.client.Do(req)
+	if err == nil && resp.StatusCode == http.StatusTemporaryRedirect {
+		loc, err := resp.Location()
+		if err != nil {
+			return nil, err
+		}
+
+		fb.url = strings.Split(loc.String(), "/.json")[0]
```

Comment:
Trying the TrimSuffix fails because the Location header includes the ?auth=thing. Since it expects a specific string, it fails to trim anything.